fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/provider/impl/FileProvider.java

```diff
@@ -0,0 +1,153 @@
+    private String source = null;
+    private long parseFailures = 0l;
+    private ImportTransientUserData transientUserData = null;
+    private List<Resource> resources = new ArrayList<>();
+    private String fhirResourceType = null;
+
+    private int total = 0;
+
+    private TransientUserData chunkData = null;
+
+    private OutputStream out = null;
+
+    private ConfigurationAdapter configuration = ConfigurationFactory.getInstance();
+
+    public FileProvider(String source) throws Exception {
+        this.source = source;
+    }
+
+    private String getFilePath(String workItem) {
+        return configuration.getBaseFileLocation(source) + "/" + workItem;
+    }
+
+    @Override
+    public long getSize(String workItem) throws FHIRException {
+        // This may be error prone as the file itself may be compressed or on a compressed volume.
+        try {
+            return Files.size(new File(getFilePath(workItem)).toPath());
+        } catch (IOException e) {
+            throw new FHIRException("Files size is not computable '" + workItem + "'", e);
+        }
+    }
+
+    @Override
+    public void readResources(long numOfLinesToSkip, String workItem) throws FHIRException {
+        try {
+            parseFailures = BulkDataUtils.readFhirResourceFromLocalFile(getFilePath(workItem), (int) numOfLinesToSkip, resources, transientUserData);
+        } catch (Exception e) {
+            throw new FHIRException("Unable to read from Local File", e);
+        }
+    }
+
+    @Override
+    public List<Resource> getResources() throws FHIRException {
+        return resources;
+    }
+
+    @Override
+    public long getNumberOfParseFailures() throws FHIRException {
+        return parseFailures;
+    }
+
+    @Override
+    public void registerTransient(ImportTransientUserData transientUserData) {
+        this.transientUserData = transientUserData;
+    }
+
+    @Override
+    public long getNumberOfLoaded() throws FHIRException {
+        return this.resources.size();
+    }
+
+    @Override
+    public void registerTransient(long executionId, TransientUserData transientUserData, String cosBucketPathPrefix, String fhirResourceType,
+        boolean isExportPublic) throws Exception {
+        if (transientUserData == null) {
+            logger.warning("registerTransient: chunkData is null, this should never happen!");
+            throw new Exception("registerTransient: chunkData is null, this should never happen!");
+        }
+
+        this.chunkData = transientUserData;
+        this.fhirResourceType = fhirResourceType;
+
+        String fileName = cosBucketPathPrefix + "_" + fhirResourceType + "_" + chunkData.getUploadCount() + ".ndjson";
+        String base = configuration.getBaseFileLocation(source);
+
+        String fn = base + "/" + fileName;
+        Path p1 = Paths.get(fn);
+        try {
+            // This is a trap. Be sure to mark CREATE and APPEND.
+            // We do create empty files.
```

Comment:
under what conditions do we create empty files?

Reply:
0 resourced files. 
E.g. export all resources and we have zero for parameters. 

I decided too put it in writeResources.

Reply:
I just changed this slightly so we don't create empty files.

Reply:
sounds good